lib/source-loader/src/abstract-syntax-tree/traverse-helpers.js

```diff
@@ -0,0 +1,203 @@
+  const metaDeclaration =
+    ast &&
+    ast.program &&
+    ast.program.body &&
+    ast.program.body.find(
+      d =>
+        d.type === 'ExportDefaultDeclaration' &&
+        d.declaration.type === 'ObjectExpression' &&
+        (d.declaration.properties || []).length
+    );
+
+  const titleProperty =
+    metaDeclaration &&
+    metaDeclaration.declaration &&
+    metaDeclaration.declaration.properties.find(p => p.key && p.key.name === 'title');
+
+  if (!titleProperty) {
+    return { addsMap, idsToFrameworks };
+  }
+  const title = titleProperty.value.value;
+
+  estraverse.traverse(ast, {
+    fallback: 'iteration',
+    enter: node => {
+      patchNode(node);
+      if (
+        node.type === 'ExportNamedDeclaration' &&
+        node.declaration &&
+        node.declaration.declarations &&
+        node.declaration.declarations.length === 1 &&
+        node.declaration.declarations[0].type === 'VariableDeclarator' &&
+        node.declaration.declarations[0].id &&
+        node.declaration.declarations[0].id.name &&
+        node.declaration.declarations[0].init &&
+        ['CallExpression', 'ArrowFunctionExpression', 'FunctionExpression'].includes(
+          node.declaration.declarations[0].init.type
+        )
+      ) {
+        let storyName = node.declaration.declarations[0].id.name;
+        storyName = findStoryTitle(storyName, ast) || storyName;
+        const toAdd = handleExportedName(title, storyName, node.declaration.declarations[0].init);
+        Object.assign(addsMap, toAdd);
+      }
+    },
+  });
+  return { addsMap, idsToFrameworks };
+}
+
+export function findDependencies(ast) {
+  const dependencies = [];
+  const storiesOfIdentifiers = {};
+
+  estraverse.traverse(ast, {
+    fallback: 'iteration',
+    enter: node => {
+      patchNode(node);
+
+      if (node.type === 'ImportDeclaration') {
+        const candidateSpecifier = (node.specifiers || []).find(
+          s => (s.imported || {}).name === 'storiesOf'
+        );
+        if (node.source.value.startsWith('@storybook/') && candidateSpecifier) {
+          Object.assign(storiesOfIdentifiers, {
+            [candidateSpecifier.local.name]: node.source.value,
+          });
+        }
+        dependencies.push(node.source.value);
+      }
+    },
+  });
+  return { dependencies, storiesOfIdentifiers };
+}
+
+export function popParametersObjectFromDefaultExport(source, ast) {
+  let splicedSource = source;
+  let parametersSliceOfCode = '';
+  let indexWhereToAppend = -1;
+  estraverse.traverse(ast, {
+    fallback: 'iteration',
+    enter: node => {
+      patchNode(node);
+
+      if (
+        node.type === 'ExportDefaultDeclaration' &&
+        node.declaration.type === 'ObjectExpression' &&
+        (node.declaration.properties || []).length
+      ) {
+        const parametersProperty = node.declaration.properties.find(
+          p => p.key.name === 'parameters' && p.value.type === 'ObjectExpression'
+        );
+
+        splicedSource =
```

Comment:
@libetl I think the error is here when there are no parameters attached to the default export